src/main/java/com/myhome/services/springdatajpa/CommunityAmenitySDJpaService.java

```diff
@@ -43,4 +46,27 @@ public boolean deleteAmenity(String amenityId) {
         .orElse(new HashSet<>());
   }
 
+  @Override
+  public boolean updateAmenity(String amenityId, CommunityAmenityDto updatedCommunityAmenity) {
+    return communityAmenityRepository.findByAmenityId(amenityId)
+      .map(communityAmenity -> communityRepository.findByCommunityId(updatedCommunityAmenity.getCommunityId())
```

Comment:
I tried using multiple .map() with `@With` annotation i.e. .map(updated -> updated.withXXX(xxx)) but some reason doing it this way would make the id of the base entity null and therefore Spring would try and insert it rather than update it, causing a unique key violation on amenity id

Reply:
Can you paste here your code snippet using map?

Reply:
I have already removed that code unfortunately, I could just call the constructor and put all the parameters into it. Or would adding @With to ID in base entity work and then call withID(..) but that would require casting?